python/ray/data/datasource/csv_datasource.py

```diff
@@ -40,19 +40,27 @@ def _read_stream(
         if hasattr(parse_options, "invalid_row_handler"):
             parse_options.invalid_row_handler = parse_options.invalid_row_handler
 
-        reader = csv.open_csv(
-            f, read_options=read_options, parse_options=parse_options, **reader_args
-        )
-        schema = None
-        while True:
-            try:
-                batch = reader.read_next_batch()
-                table = pyarrow.Table.from_batches([batch], schema=schema)
-                if schema is None:
-                    schema = table.schema
-                yield table
-            except StopIteration:
-                return
+        try:
+            reader = csv.open_csv(
+                f, read_options=read_options, parse_options=parse_options, **reader_args
+            )
+            schema = None
+            while True:
+                try:
+                    batch = reader.read_next_batch()
+                    table = pyarrow.Table.from_batches([batch], schema=schema)
+                    if schema is None:
+                        schema = table.schema
+                    yield table
+                except StopIteration:
+                    return
+        except Exception as e:
+            raise type(e)(
+                f"{e}. Failed to read CSV file: {path}. "
+                "Please check the file has correct format, or filter out file with "
```

Comment:
So at this point, is it sure that only incorrect format will fail the open_csv?

Reply:
We don't know for sure. Arrow just returns a generic `pyarrow.lib.ArrowInvalid: CSV parse error`, so it can also because the CSV file is malformed.

Changed to catch `pyarrow.lib.ArrowInvalid` only, and printed out error message to try to be reasonable in all cases:

```
pyarrow.lib.ArrowInvalid: CSV parse error: Row #2: Expected 2 columns, got 3: $$&�5one��&R&(, ...
Failed to read CSV file: ...
Please check the CSV file has correct format, or filter out non-CSV file with 'partition_filter' field.
See read_csv() documentation for more details.
```